services/src/datasets/create_from_workflow.rs

```diff
@@ -0,0 +1,268 @@
+    info: RasterDatasetFromWorkflow,
+) -> error::Result<TaskId> {
+    let upload = UploadId::new();
+    let upload_path = upload.root_path()?;
+    fs::create_dir_all(&upload_path)
+        .await
+        .context(crate::error::Io)?;
+    let file_path = upload_path.clone();
+
+    let task = RasterDatasetFromWorkflowTask {
+        workflow,
+        session,
+        ctx: ctx.clone(),
+        info,
+        upload,
+        file_path,
+    }
+    .boxed();
+
+    let task_id = ctx.tasks_ref().schedule(task, None).await?;
+
+    Ok(task_id)
+}
+
+async fn create_dataset<C: Context>(
+    info: RasterDatasetFromWorkflow,
+    mut slice_info: Vec<GdalLoadingInfoTemporalSlice>,
+    origin_result_descriptor: &RasterResultDescriptor,
+    query_rectangle: RasterQueryRectangle,
+    ctx: &C,
+    session: <C as Context>::Session,
+) -> error::Result<DatasetId> {
+    ensure!(!slice_info.is_empty(), error::EmptyDatasetCannotBeImported);
+
+    let dataset_id = DatasetId::new();
+    let first_start = slice_info
+        .first()
+        .expect("slice_info should have at least one element")
+        .time
+        .start();
+    let last_end = slice_info
+        .last()
+        .expect("slice_info should have at least one element")
+        .time
+        .end();
+    let result_time_interval = TimeInterval::new(first_start, last_end)?;
+
+    let result_descriptor = RasterResultDescriptor {
+        data_type: origin_result_descriptor.data_type,
+        spatial_reference: origin_result_descriptor.spatial_reference,
+        measurement: origin_result_descriptor.measurement.clone(),
+        time: Some(result_time_interval),
+        bbox: Some(query_rectangle.spatial_bounds),
+        resolution: Some(query_rectangle.spatial_resolution),
+    };
+    //TODO: Recognize MetaDataDefinition::GdalMetaDataRegular
+    let meta_data = if slice_info.len() == 1 {
+        let loading_info_slice = slice_info.pop().expect("slice_info has len one");
+        let time = Some(loading_info_slice.time);
+        let params = loading_info_slice
+            .params
+            .expect("datasets with exactly one timestep should have data");
+        MetaDataDefinition::GdalStatic(GdalMetaDataStatic {
+            time,
+            params,
+            result_descriptor,
+        })
+    } else {
+        MetaDataDefinition::GdalMetaDataList(GdalMetaDataList {
+            result_descriptor,
+            params: slice_info,
+        })
+    };
+
+    let dataset_definition = DatasetDefinition {
+        properties: AddDataset {
+            id: Some(dataset_id),
+            name: info.name,
+            description: info.description.unwrap_or_default(),
+            source_operator: "GdalSource".to_owned(),
+            symbology: None,  // TODO add symbology?
+            provenance: None, // TODO add provenance that references the workflow
+        },
+        meta_data,
+    };
+
+    // TODO: build pyramides, prefereably in the background
```

Comment:
pyramides are build when data is stored as COG. I guess we don't need to do that as an additional task.

Reply:
d9b7cbc73daa7c5b2990340dd37619fc3debdcac